src/main/java/seedu/address/model/person/RoomNumber.java

```diff
@@ -38,14 +42,43 @@ public class RoomNumber {
      * @param roomNumber A valid room number.
      */
     public RoomNumber(String roomNumber) {
+        this(roomNumber, LocalDate.now());
+    }
+
+    /**
+     * Constructs an {@code RoomNumber}.
+     *
+     * @param roomNumber A valid room number.
+     * @param lastModified A valid room number.
+     */
+    public RoomNumber(String roomNumber, LocalDate lastModified) {
         requireNonNull(roomNumber);
         checkArgument(isValidRoomNumber(roomNumber), MESSAGE_CONSTRAINTS);
         Matcher matcher = Pattern.compile(VALIDATION_REGEX).matcher(roomNumber);
         matcher.find();
         this.block = matcher.group(1);
         this.floor = matcher.group(2);
         this.roomNumber = matcher.group(3);
-        lastModified = LocalDate.now();
+        this.lastModified = lastModified;
+    }
+
+    /**
+     * Constructs an {@code RoomNumber}.
+     *
+     * @param roomNumber A valid room number.
+     * @param flag Indicator to process date from roomNumber.
+     */
+    public RoomNumber(String roomNumber, boolean flag) {
+        requireNonNull(roomNumber);
+        checkArgument(isValidRoomNumberWDate(roomNumber), MESSAGE_CONSTRAINTS_DATE);
+        Matcher matcher = Pattern.compile(VALIDATION_REGEX_W_DATE).matcher(roomNumber);
+        matcher.find();
+        LocalDate date = LocalDate.parse(matcher.group(4));
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS_DATE);
+        this.block = matcher.group(1);
+        this.floor = matcher.group(2);
+        this.roomNumber = matcher.group(3);
+        this.lastModified = date;
     }
```

Comment:
May I ask what is the purpose of this function? Looks like the flag is not used

Reply:
Yea the flag is kinda like a dummy variable to indicate that the date is in the roomNumber string. This is for JSON / any backend compatibility. It means that the RoomNumber class can handle the regex instead of asking any calling class to do it first.

Reply:
Interesting, do you mean the purpose of the flag parameter is just to overload the constructor, and other files will be using this function?

Reply:
Yep correct. If the String has a date in it use the flag. If not use the regular function.